src/VisualStudio/Core/Def/Implementation/EditAndContinue/VsENCRebuildableProjectImpl.cs

```diff
@@ -980,6 +741,18 @@ public int ExitBreakStateOnPE()
             }
         }
 
+        private static ImmutableDictionary<K, ImmutableArray<V>> GroupToImmutable<K, V>(IEnumerable<IGrouping<K, V>> items)
+        {
+            var builder = ImmutableDictionary.CreateBuilder<K, ImmutableArray<V>>();
+
+            foreach (var item in items)
+            {
+                builder.Add(item.Key, item.ToImmutableArray());
+            }
+
+            return builder.ToImmutable();
+        }
```

Comment:
Do you want to move this to extensions as well?

Reply:
I don't see a better fit for this to be in. I think we need to clean up our extensions first, but that's for another day.

---
In reply to: [162476175](https://github.com/dotnet/roslyn/pull/24316#discussion_r162476175) [](ancestors = 162476175)